### ISSUE TITLE
wagtail-localize: make all snippets translatable

### ISSUE THREAD

**Issue by TheoChevalier** [mozilla/foundation.mozilla.org#6059]
Unlike models, snippets are not translatable by default. We need to make small adjustments to each of them:

- [ ] Add the `TranslatableMixin` mixin
- [ ] List all fields that need localization to the `translatable_fields` variable
- [ ] Optionally, add fields to `override_translatable_fields` (e.g. ShareProgress fields need to be added to both variables, so that they can be overriden in the admin, but don’t end up in .po files)
- [ ] Probably do something around `unique_together`

Example of translatable snippet: https://github.com/wagtail/wagtail-localize/blob/fdd813e8a9717b8336f31ba9d04b75976d1d915f/wagtail_localize/test/models.py#L200-L216

**Comment by Pomax** [comment]
also tagging @kaedroho as this may need some professional expert advice!

**Comment by KalobTaulien** [comment]
Moving this to blocked because of a bug when a model inherits from another model as a registered snippet. I opened an issue here: https://github.com/wagtail/wagtail-localize/issues/321 - I've also let Karl know about this.

**Comment by KalobTaulien** [comment]
Update: This is a wagtail core bug that should hopefully be released in the next point release upgrade. I might take a run at it with the latest wagtail code and latest wagtail-localize code just to make sure. If I do that, I'll report my findings back here.

**Comment by KalobTaulien** [comment]
Back to blocked. Too many packages are pinned to wagtail 2.12 but `master` is at `2.13a0`. Will need to come back to this when localize, footnotes, and other wagtail packages are accepting 2.13 soon.

**Comment by benhohner** [comment]
Kalob to investigate since version bumped

**Comment by KalobTaulien** [comment]
https://github.com/wagtail/wagtail/pull/6816 is merged into wagtail core. Along with Wagtail 2.11.16 version bump. I'll test this out with the new wagtail version on Monday.

**Comment by KalobTaulien** [comment]
This is almost unblocked. We just need to merge in #6340 for a Wagtail 2.11.6 version bump to solve the solution, and then we can commence work on this ticket again.
